### PR TITLE
SageIntacct API: Workspace App Cleanup - Rename, Refactor, Remove, Documentation

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/sage_intacct/helpers.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from apps.sage_intacct.tasks import schedule_ap_payment_creation, schedule_sage_intacct_objects_status_sync, \
+    schedule_sage_intacct_reimbursement_creation
+from apps.workspaces.models import Configuration
+
+def schedule_payment_sync(configuration: Configuration):
+    """
+    :param configuration: Workspace Configuration Intance
```
Comment (ashwin1111) [inline apps/sage_intacct/helpers.py:7]: Instance

File: apps/workspaces/permissions.py [hunk 2]
```diff
@@ -11,15 +11,13 @@ class WorkspacePermissions(permissions.BasePermission):
...
-
-    def validate_and_cache(self, workspace_users, user: User, workspace_id: str, cache_users: bool = False):
+    @staticmethod
+    def validate_and_cache(workspace_users, user: User, workspace_id: str, cache_users: bool = False):
         if user.id in workspace_users:
             if cache_users:
                 cache.set(workspace_id, workspace_users, 172800)
             return True
         
-        return False
```
Comment (ashwin1111) [inline apps/workspaces/permissions.py:21]: why is this removed?

File: apps/workspaces/serializers.py [hunk 3]
```diff
@@ -15,12 +15,53 @@ class Meta:
...
+        """
+        Create Workspace General Settings
+        :param validated_data: Validated data
+        :return: upserted general settings object
+        """
+        workspace = validated_data['workspace']
+
+        configuration, _ = Configuration.objects.update_or_create(
+            workspace_id=workspace,
+            defaults={
```
Comment (ashwin1111) [inline apps/workspaces/serializers.py:34]: why not just pop workspace from validated_data and use the variable here?

File: apps/workspaces/urls.py [hunk 4]
```diff
@@ -15,23 +15,38 @@
...
     path('<int:workspace_id>/tasks/', include('apps.tasks.urls')),
-    path('ready/', ReadyView.as_view({'get': 'get'})),
-    path('<int:workspace_id>/schedule/', ScheduleView.as_view({'post': 'post', 'get': 'get'}))
 ]
+
+urlpatterns = []
+urlpatterns.extend(workspace_app_paths)
+urlpatterns.extend(fyle_connection_api_paths)
+urlpatterns.extend(sage_intacct_connection_api_paths)
+urlpatterns.extend(other_app_paths)
```
Comment (ashwin1111) [inline apps/workspaces/urls.py:52]: new line

File: apps/workspaces/utils.py [hunk 5]
```diff
@@ -1,70 +0,0 @@
...
-    if general_settings.corporate_credit_card_expenses_object == 'CHARGE_CARD_TRANSACTION':
-        expense_group_settings = ExpenseGroupSettings.objects.get(workspace_id=workspace_id)
-
-        ccc_expense_group_fields = expense_group_settings.corporate_credit_card_expense_group_fields
-        ccc_expense_group_fields.append('expense_id')
-        expense_group_settings.corporate_credit_card_expense_group_fields = list(set(ccc_expense_group_fields))
-
-        expense_group_settings.save()
-
-    schedule_categories_creation(import_categories=general_settings.import_categories, workspace_id=workspace_id)
```
Comment (ashwin1111) [inline apps/workspaces/utils.py:50]: Double check if you've covered all post triggers

File: requirements.txt [hunk 6]
```diff
@@ -3,7 +3,7 @@ dj-database-url==0.5.0
 Django==3.0.3
 django-cors-headers==3.2.0
 django-picklefield==3.0.1
-django-q==1.3.3
+django-q==1.3.4
```
Comment (ashwin1111) [inline requirements.txt:6]: why is this changed? Make sure there are no bad migrations

Comment (NileshPant1999) [reply requirements.txt:6]: 1.3.3 was giving error on new version of mac so i locally updated it if you want me roll back i can do it

File: apps/mappings/helpers.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+from apps.mappings.tasks import schedule_projects_creation, schedule_categories_creation, schedule_auto_map_employees, \
```
Comment (Sravanksk) [inline apps/mappings/helpers.py:1]: `schedule_projects_creation` this import not being used

File: requirements.txt (hunk 6 above)
Comment (ashwin1111) [reply requirements.txt:6]: what's the error?

**Review by ashwin1111 — APPROVED** [review]

File: requirements.txt (hunk 6 above)
Comment (NileshPant1999) [reply requirements.txt:6]: it was related to pickle

Comment (NileshPant1999) [reply requirements.txt:6]: rolling it back

**Review by Sravanksk — APPROVED** [review]
